apm-lambda-extension/extension/process_events.go

```diff
@@ -34,6 +34,9 @@ func FlushAPMData(client *http.Client, dataChannel chan AgentData, config *exten
 	for {
 		select {
 		case agentData := <-dataChannel:
+			if !IsTransportStatusHealthy() {
+				return
+			}
```

Comment:
If we know at the beginning of the method that the status is not healthy, we don't have to remove any agent data from the channel
```suggestion
	if !IsTransportStatusHealthy() {
		return
	}
	for {
		select {
		case agentData := <-dataChannel:
```

Reply:
Solved in ad48058